Add issue endpoint

YukinaMochizuki opened the pull request:
## Type of changes
Feature

## Purpose
### Add `Issue` reated endpoints
- Get all `Issues` with pagination
- Get `Issue` by id
- Get all `Facts` that all `Viewpoints` used in `Issue`

### Add manual test endpoint
- Create `Issue`
- Update `Issue`
- Delete `Issue`

## Additional information
- Define the "事件簡述" from the Figma design as the field "insight" in `Issue` schema

burnedinthesky reviewed (changes requested):
@YukinaMochizuki 
Thanks for the nice work! The API looks quite mature, however, there are some concerns and questions I would like to bring up.

> GET `/issues`

From the looks of the sample data, it seems like all facts related to the issue would be sent along with the issue. I believe this approach may result in an extremely large payload (since it is likely for an issue to have hundreds if not thousands of facts), which is also redundant (as the only page which consumes the endpoint, the home page, does not display any fact related data). Thus, I propose we remove the facts field from the `/issues` endpoint.

> Lack of GET `/issues/{id}` endpoint

When rendering `commonground.tw/issues/{id}`, the frontend team will need an endpoint that exclusively returns issue data corresponding to that specific ID.

> Inconsistent plurality in "GET `/issues/{id}/facts`" and "GET | POST `/issue/{id}/viewpoints`"

The former is issues (plural) while the latter is issue (single). Is this difference in plurality by design?

> Lack of POST `/issue/{id}/facts`

I believe there should be an endpoint corresponding to the issue's facts' page "create new fact". Since I don't see an `issue` field in "POST `/api/facts`", I'm assuming there should be another endpoint to correlate facts with issues. 


Lastly, just to confirm, I see two versions of endpoints "POST `/issues/`" and "PUT | DELETE `/issues/{id}`" in both sections `Issue` and `Admin`. Is this a feature of swagger, or were the endpoints duplicated?

Furthermore, all newly created endpoints don't have the `/api` prefix before their paths? Is this intentional?

All other aspects look good to me. Thanks for the hard work!

YukinaMochizuki commented:
Thank you for your review! I have fixed some typos, such as the missing /api/ and incorrect plurals.

> From the looks of the sample data, it seems like all facts related to the issue would be sent along with the issue. I believe this approach may result in an extremely large payload ...

The `Facts` field in the `Issue` contains only the `Facts` referenced in the `Insight`. Unless our `Insight` references a large number of `Facts`, there shouldn’t be significant performance issues? I expect that in the future, the `Facts` within `Insight` might have a preview feature, such as displaying a preview when hovered over. This design choice is aimed at optimizing the Round-trip Delay Time issue, which we all care about.

Additionally, this reminded me to pay attention to the First Contentful Paint (FCP) metric. I added `SimpleIssue` for use with GET `/api/issues`:
```yaml
SimpleIssue:
  id:
    type: string
    format: uuid
  createdAt:
    type: string
    format: date-time
  updatedAt:
    type: string
    format: date-time
    description: The last time the issue title, description, or insight was updated
  title:
    type: string
  description:
    type: string
```

> two versions of endpoints "POST /issues/" and "PUT | DELETE /issues/{id}" in both sections Issue and Admin. Is this a feature of swagger, or were the endpoints duplicated?

This is a feature of Swagger. By default, Swagger categorizes and displays each endpoint by tags, which means if an endpoint has more than one tag, it will appear in multiple places.

### Other Changes
- Add POST `/api/issue/{id}/facts`
- Add GET `/api/issues/{id}`

burnedinthesky reviewed (approved):
@YukinaMochizuki  Thanks for your response! Everything now looks good to me. I would like to confirm something though.

> The Facts field in the Issue contains only the Facts referenced in the Insight. 

I did not realize that was the case. In this scenario I think this design makes sense and is quite smart, thanks for clarifying. In light of this clarification, I would like to ask whether the `facts` fiels in "POST `/api/issues`" and "PUT `/api/issues/{id}`" indicates an initial set of facts that should be related to this issue immediately, or if it is the facts referenced by the issue summary.
